api/src/repositories/survey-block-repository.ts

```diff
@@ -0,0 +1,147 @@
+import SQL from 'sql-template-strings';
+import { z } from 'zod';
+import { ApiExecuteSQLError } from '../errors/api-error';
+import { BaseRepository } from './base-repository';
+
+export const SurveyBlock = z.object({
+  survey_block_id: z.number().nullable(),
```

Comment:
Should we remove this property? Is there ever a case where you expect `SurveyBlock.survey_block_id` to be defined, but couldn't otherwise just use `SurveyBlockRecord` instead?

Reply:
`survey_block_id` is used to check if an incoming block needs to be updated or created net new.

Reply:
But if a block has a `survey_block_id`, then we can also assert it has all of the properties that a `SurveyBlockRecord` has, no?

Reply:
I left the zod schema alone and decided to just rename the SurveyBlock to PostSurveyBlock to indicate it is for inserting data. I also turned it into an interface because there was no reason for it to be a zod schema.